### PR TITLE
lint check

### PR DESCRIPTION
**Pull request by hannesa2**
To improve code quality, I recommend a lint check on every pull request. 
Currently it's obvious disabled, because on master it fails ! **Please enable lint check on CI** 

With this pull request you see in root the lint report `lint-app-report.html`
image

Btw, a lot of them should be solved later on, and after that step, you should prevent them in the future by `severity="error"` in _lint.xml_

### CODE REVIEW

**Comment by davigonz** [comment]
Thanks @hannesa2 , this is a good step after merging [the modularization](https://github.com/owncloud/android/issues/2607) into master

**Comment by hannesa2** [comment]
Why not now ?

**Comment by davigonz** [comment]
> Why not now ?

Many changes have taken place during modularization, a bunch of new files have been created, old files have been removed. If we move forward the lint check before merging modularization we could have lint suggestions that are already solved. 

I'd rather include lint check and start fixing the suggestions after the huge structural change of modularization. Besides, the modularization merge into master should not take long, the last PR is ready to code review => https://github.com/owncloud/android/pull/2728

**Comment by hannesa2** [comment]
Interesting point of view.
To resume it: This PR has no blocking lint issues, I solved all blockers in this PR !

Knowing this, there are two ways:
A: merging this PR **now**, and **you have**  to solve **your** hypothetical lint by yourself
B: merging this PR **after** your merge, and **I've** to solve **your** hypothetical lint by me in this PR

To be honest, that's cheeky

**Comment by davigonz** [comment]
> A: merging this PR now, and you have to solve your hypothetical lint by yourself
> B: merging this PR after your merge, and I've to solve your hypothetical lint by me in this PR

1. Please stop using sarcasm and these **your** , **I've** accusations, are disrespectful and being respectful is one of the main points of our [community code of conduct](https://owncloud.org/community/code-of-conduct/). Besides, using "cheeky" is out of place here, specially when some of your PRs have not been finished by yourself - applying requested changes or fixing reported bugs and that's what every contributor or developer we have had has always done with no complains - and we have helped you to move them forward, last example => https://github.com/owncloud/android/pull/2725.

2. No one said you have to solve the lint suggestions in this PR after merging modularization, sorry if I did not explain myself properly. My idea was taking advantage of this PR and applying the lint suggestions of modularization here by ourselves, in the different modules and so on... addressing all the work related to the lint here and not splitting it up into different issues.

To sum up, I do not want to delay more the modularization merge into master and merging this PR before will block it since the modularization build will fail because of lint errors (see below) => no merge possible. 

Screenshot 2019-12-02 at 12 59 44

Therefore, just after merging the modularization into master, we will enable the lint check via this PR y start to work on it (I will do it, no problem).

File: .gitignore [hunk 1]
```diff
@@ -8,19 +8,11 @@
 # Java class files
 *.class
 
-# generated files
-bin/
-build/
-*.iml
-gen/
-target/
-
```
Comment (abelgardep) [inline .gitignore]: Why do you remove this generated files from gitignore?

File: .gitignore [hunk 2]
```diff
@@ -33,13 +25,15 @@ tests/proguard-project.txt
 
 # Android Studio and Gradle specific entries
 .gradle
-.idea
```
Comment (abelgardep) [inline .gitignore]: Why do you remove this?

File: owncloudApp/lint.xml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<lint>
+    <issue id="WrongConstant" severity="ignore" />
+    <issue id="MissingTranslation" severity="ignore" />
+    <issue id="ByteOrderMark" severity="ignore" />
```
Comment (abelgardep) [inline owncloudApp/lint.xml]: Which criteria do you follow to ignore this lints?

File: .gitignore (hunk 1 above)
Comment (hannesa2) [reply .gitignore]: bin, gen and target is Eclipse
build in in L38 too
.iml in in L37 too

File: .gitignore (hunk 2 above)
Comment (hannesa2) [reply .gitignore]: [L46-L48](https://github.com/owncloud/android/blob/master/.gitignore#L43-L45) has more relevance, there is all excluded, except code format

File: owncloudApp/lint.xml (hunk 3 above)
Comment (hannesa2) [reply owncloudApp/lint.xml]: Without them, it will fail.
Everyone is welcome to remove this, and solve this, after a merge

Comment (hannesa2) [reply owncloudApp/lint.xml]: Please see the post for @davigonz 
image

Comment (abelgardep) [reply owncloudApp/lint.xml]: Ok, thanks, i'll check it in #2750

**Comment by abelgardep** [comment]
Will be continued on #2750
